src/main/java/com/samoilenko/onlinebookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -52,6 +57,16 @@ protected ResponseEntity<Object> handleRegistrationException(
             RegistrationException ex, WebRequest request
     ) {
         String responseMessage = "Can't register user: " + ex.getMessage();
+        log.error("Can't register user", ex);
+        return getResponseEntity(HttpStatus.BAD_REQUEST, responseMessage);
+    }
+
+    @ExceptionHandler(SQLIntegrityConstraintViolationException.class)
```

Comment:
in a perfect application we would prevent all of SQL and DB integrity related problems by doing all the necessary checks and validations **before** sending the actual save request to DB.
Treat SQL exceptions like some NPE, if we encounter it - we should prevent this situation from happening and not try/catch it :)